diff --git a/editparams.cgi b/editparams.cgi
index 28aac96..225c995 100755
--- a/editparams.cgi
+++ b/editparams.cgi
@@ -128,6 +128,8 @@ if ($action eq 'save' && $current_module) {
         }
 
         if ($changed) {
+            # Hook to test value
+            Bugzilla::Hook::process('check_parameter', { name => $name, value => $value });
             if (exists $i->{'checker'}) {
                 my $ok = $i->{'checker'}->($value, $i);
                 if ($ok ne "") {
diff --git a/template/en/default/bug/field.html.tmpl b/template/en/default/bug/field.html.tmpl
index 9c956ea..c534bf7 100644
--- a/template/en/default/bug/field.html.tmpl
+++ b/template/en/default/bug/field.html.tmpl
@@ -165,6 +165,7 @@
            <label><input type="checkbox" value="[% bug_url.name FILTER html %]"
                          name="remove_[% field.name FILTER html %]">
              Remove</label>
+           [% Hook.process('enhancedseealso') %]
          </li>
        [% END %]
        [% '</ul>' IF value.size %]
diff --git a/template/en/default/list/table.html.tmpl b/template/en/default/list/table.html.tmpl
index 2b266d4..2d543fb 100644
--- a/template/en/default/list/table.html.tmpl
+++ b/template/en/default/list/table.html.tmpl
@@ -194,6 +194,8 @@
     [% FOREACH column = displaycolumns %]
     <td [% 'style="white-space: nowrap"' IF NOT abbrev.$column.wrap %]
         class="bz_[% column FILTER css_class_quote %]_column">
+[% Hook.process('field_start') %]
+
       [% IF abbrev.$column.maxlength %]
         <span title="[%- display_value(column, bug.$column) FILTER html %]">
       [% END %]
@@ -218,6 +220,7 @@
       [% IF abbrev.$column.maxlength %]
         </span>
       [% END %]
+[% Hook.process('field_end') %]
     </td>
     [% END %]
 
